### PR TITLE
Add qualifications and program type to UCAS export

### PR DESCRIPTION
**Pull request by aldavidson**
## Context
Please note: migrations are included in a separate PR #2008 - please merge that one _first_

As part of ongoing discussions around the data sharing agreement, UCAS have asked for the fields `qualifications` and `program_type` from the Find API to be included in the 'Dataset1' file which we will generate for them each night. These fields are pre-emptively documented in the [Find API Docs](https://api2.publish-teacher-training-courses.service.gov.uk/api-reference.html#schema-courseattributes) but are not yet implemented in the actual API itself.

So we have agreed to add the fields into the export file now, so that they can integrate on the basis that the fields are there, and we will make them blank values until the fields are present in the Find API (ETA: "a few weeks") at which time the values will appear with no further code changes needed.

## Changes proposed in this pull request

1. Populate new fields in `SyncProviderFromFind` 
2. Include fields in `ApplicationsExportForUCAS` service, renaming what was labelled as "Programme Type" back to match our original name "funding_type".
3. Add expectations in the relevant specs for the above

## Guidance to review

Make sure you have some submitted applications in your DB (run `GenerateTestData.new(10).generate` from the console if not)

Then, from the support UI:
1. Enable the `download_dataset1_from_support_page` FeatureFlag
2. Go to /support/performance
3. Click on 'Download Dataset1 (CSV)'
4. View the downloaded file - you should see "Program type" and "Qualifications" fields in the header. All values for those fields should be empty.

## Link to Trello card

[Add program_type and qualifications to the UCAS dataset1 export, (with blank values until Find add them to the API)](https://trello.com/c/igGCClZc/1426-add-programtype-and-qualifications-to-the-ucas-dataset1-export-with-blank-values-until-find-add-them-to-the-api)

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/applications_export_for_ucas.rb [hunk 1]
```diff
@@ -82,25 +83,23 @@ def convert_to_hash(application_choice)
...
   end
 
-  def subject_codes(course)
-    course.subject_codes.to_a.join('|')
-  end
-
-  def disabilities(form)
-    form.equality_and_diversity.try(:[], 'disabilities').to_a.join('|')
+  def concatenate(array)
+    array.to_a.join('|')
```
Comment (tvararu) [inline app/services/applications_export_for_ucas.rb:103]: Assuming `.to_a` works on JSONB? LGTM then 👍

File: spec/support/test_helpers/find_api_helper.rb [hunk 2]
```diff
@@ -462,6 +462,137 @@ def stub_find_api_course(provider_code, course_code)
       "/courses/#{course_code}")
   end
 
+  def stub_find_api_provider_200_with_qualifications_and_program_type(
```
Comment (tvararu) [inline spec/support/test_helpers/find_api_helper.rb:465]: Not sure if this needs to be in `spec/support/test_helpers/find_api_helper.rb`. Can it live in the only test file that currently uses it, `spec/services/sync_provider_from_find_spec.rb`?

Edit: I see there's a few more such methods in this file, so you're following an established pattern. In that case, do it this way, and if it gets unwieldy we can revisit and refactor. 👍

**Comment by aldavidson** [comment]
> Can you implement an enum for the possible values of `program_type`?
> 
> Screenshot 2020-05-06 at 12 42 06
> 
> See [ae40c26](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/commit/ae40c26f19447cb25af70e8302edc065cb3c3190) for Duncan's approach for doing this to `funding_type`

Done!

**Comment by aldavidson** [comment]
@tvararu ready for re-review with the `enum` added, whenever you are

**Pull request merged**